test/functional/test_framework/authproxy.py

```diff
@@ -78,7 +78,9 @@ def __init__(self, service_url, service_name=None, timeout=HTTP_TIMEOUT, connect
         passwd = None if self.__url.password is None else self.__url.password.encode('utf8')
         authpair = user + b':' + passwd
         self.__auth_header = b'Basic ' + base64.b64encode(authpair)
-        self.timeout = timeout
+        # larger values of the timeout cause an "Invalid argument"
+        # socket exception in Python's HTTP client library, so clamp it
+        self.timeout = min(timeout, 2147483)
```

Comment:
Out of curiosity, is there a reason you chose `2147483`?

Reply:
That's the max 31-bit number, converted from milliseconds to seconds.

But this is assuming the OS has specific limitations. Not really a good fix.

Reply:
Got it, thanks Luke.